src/managed/Microsoft.Extensions.DependencyModel/JsonTextReaderExtensions.Netstandard2_0.cs

```diff
@@ -0,0 +1,139 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text.Json;
+
+namespace Microsoft.Extensions.DependencyModel
+{
+    internal static class JsonTextReaderExtensions
```

Comment:
It feels like all these methods are things that most consumers would want... and it is obviously a non-trivial amount of code.

Reply:
@eerhardt A lot of the code in these extension methods is just about changing the exception type,  and combining the move and get operations.  If InvalidOperationException were acceptable then `ReadAsString()` is just

```C#
reader.Next();
return reader.GetStringValue();
```

The return value of Next is important when using the reader in async/chunked mode; and an exception model is what it is...